source/common/router/config_impl.cc

```diff
@@ -418,38 +417,36 @@ void RouteEntryImplBase::finalizeResponseHeaders(
   vhost_.globalRouteConfig().responseHeaderParser().evaluateHeaders(headers, request_info);
 }
 
+bool RouteEntryImplBase::evaluateRuntimeKeys(const RuntimeData& runtime_data,
+                                             const uint64_t random_value) const {
+  envoy::type::FractionalPercent fractional_percent;
+  const std::string& fraction_yaml = loader_.snapshot().get(runtime_data.fractional_runtime_key);
+
+  try {
+    MessageUtil::loadFromYamlAndValidate(fraction_yaml, fractional_percent);
```

Comment:
This is in the data plane path, right? I think this is definitely not what we want. We need to store the key name, etc. and then do the runtime lookup?